Source/missiles.cpp

```diff
@@ -915,6 +874,105 @@ Direction16 GetDirection16(Point p1, Point p2)
 	return ret;
 }
 
+void MissileHitMonsterConsequences(int mid, int pnum, int dam, missile_id mName)
+{
+	auto &monster = Monsters[mid];
+	const auto &player = Players[pnum];
+
+	bool hasKnockback = false;
+	if (pnum >= 0) {
+		const auto &player = Players[pnum];
+		if (HasAnyOf(player._pIFlags, ItemSpecialEffect::Knockback))
+			hasKnockback = true;
+	}
+
+	if (monster._mhitpoints >> 6 <= 0) {
+		M_StartKill(mid, pnum);
+	} else if (monster.IsResistant(mName)) {
+		PlayEffect(monster, 1);
+	} else {
+		if (monster._mmode != MonsterMode::Petrified
+		    && MissilesData[mName].mType == 0
+		    && hasKnockback)
+			M_GetKnockback(mid);
+		if (monster.MType->mtype != MT_GOLEM)
+			M_StartHit(mid, pnum, dam);
+	}
+}
+
+int PlayerMissileCalculateCTHAgainstMonster(int pnum, missile_id mName, int dist, Monster &monster)
+{
+	int hper = 0;
+	if (pnum != -1) {
+		const auto &player = Players[pnum];
+		if (MissilesData[mName].mType == 0) {
+			hper = player.GetRangedPiercingToHit();
+			hper -= player.CalculateArmorPierce(monster.mArmorClass, false);
+			hper -= (dist * dist) / 2;
+		} else {
+			hper = player.GetMagicToHit() - (monster.mLevel * 2) - dist;
+		}
+	} else {
+		hper = GenerateRnd(75) - monster.mLevel * 2;
+	}
+	return hper;
+}
+
+int PlayerMissileCalculateDamageAgainstMonster(int pnum, int minDam, int maxDam, missile_id mName)
+{
+	int dam = minDam + GenerateRnd(maxDam - minDam + 1);
+
+	const auto &player = Players[pnum];
+
+	if (MissilesData[mName].mType == 0 && MissilesData[mName].mResist == MISR_NONE) {
+		dam = player._pIBonusDamMod + dam * player._pIBonusDam / 100 + dam;
+		if (player._pClass == HeroClass::Rogue)
+			dam += player._pDamageMod;
+		else
+			dam += player._pDamageMod / 2;
+	}
+	return dam;
+}
+
+void PlayerMissileHitMonster(int pnum, int dam, int m, missile_id mName)
+{
+	auto &monster = Monsters[m];
+	if (pnum == MyPlayerId)
+		monster._mhitpoints -= dam;
+
+	const auto &player = Players[pnum];

```

Comment:
```suggestion
void PlayerMissileHitMonster(const Player &player, int dam, int m, missile_id mName)
{
	Monster &monster = Monsters[m];
	if (&player == MyPlayer)
		monster._mhitpoints -= dam;

```

Reply:
I did replace auto for Monster, the parameter part is outdated. But I will try to improve it even more.

Reply:
I hope to continue the review sometime this weekend.

Reply:
this is probably resolved now

Reply:
> I hope to continue the review sometime this weekend.

 I guess I should ask which one is *this* weekend :cat:

Reply:
Yeah that is totally fair. Things have been moving pretty fast lately, and there has been more going on in my life then usual so I have been unable to keep up with it all. I'm keeping an open tab for this so it's not forgotten. I also have a note to test your bug fix PRs so we can get them merged as well :)